packages/core/tests/workers.test.ts

```diff
@@ -0,0 +1,114 @@
+      await enqueueNJobs(mysqlQueue, queueName, 10);
+
+      void Promise.all([worker1.start(), worker2.start()]);
+      await promise;
+
+      const w1JobIds = Worker1HandlerMock.handle.mock.calls.map((c) => c[0].id);
+      const w2JobIds = Worker2HandlerMock.handle.mock.calls.map((c) => c[0].id);
+      expect(haveNoCommonElements(w1JobIds, w2JobIds)).toBeTruthy();
+      expect(Worker1HandlerMock.handle).toHaveBeenCalledTimes(5);
+      expect(Worker2HandlerMock.handle).toHaveBeenCalledTimes(5);
+    });
+
+    it("should ensure that a slow worker does not block or delay other workers, case jobs already on queue", async () => {
+      await enqueueNJobs(mysqlQueue, queueName, 10);
+      void Promise.all([worker1.start(), worker2.start()]);
+      const promise = mysqlQueue.getJobExecutionPromise(queueName, 10);
+
+      await promise;
+
+      await sleep(500);
+      const [rows] = await pool.query<RowDataPacket[]>(`SELECT id, createdAt, completedAt from ${mysqlQueue.jobsTable()}`);
+      const jobs = rows.map((j) => ({ ...j, durationMs: new Date(j.completedAt).getTime() - new Date(j.createdAt).getTime() }));
+      expect(hasExactly(jobs, 5, (item) => item.durationMs > 3000)).toBeTruthy();
+      expect(hasExactly(jobs, 5, (item) => item.durationMs < 100)).toBeTruthy();
+    });
+
+    it("should ensure that a slow worker does not block or delay other workers, case no jobs on queue", async () => {
+      void Promise.all([worker1.start(), worker2.start()]);
+      await enqueueNJobs(mysqlQueue, queueName, 10);
+      const promise = mysqlQueue.getJobExecutionPromise(queueName, 10);
+
+      await promise;
+
+      await sleep(500);
+      const [rows] = await pool.query<RowDataPacket[]>(`SELECT id, createdAt, completedAt from ${mysqlQueue.jobsTable()}`);
+      const jobs = rows.map((j) => ({ ...j, durationMs: new Date(j.completedAt).getTime() - new Date(j.createdAt).getTime() }));
+      expect(hasExactly(jobs, 5, (item) => item.durationMs > 3000)).toBeTruthy();
+      expect(hasExactly(jobs, 5, (item) => item.durationMs < 1000)).toBeTruthy();
+    });
+  });
+});
+
+function sleep(ms: number) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+function haveNoCommonElements(arr1: unknown[], arr2: unknown[]) {
+  const set1 = new Set(arr1);
+  return arr2.every((item) => !set1.has(item));
+}
+
+function enqueueNJobs(mysqlQueue: MysqlQueue, queueName: string, n: number) {
+  return mysqlQueue.enqueue(
+    queueName,
+    Array.from({ length: n }).map((_, index) => ({ name: `job-${index}`, payload: {}, startAfter: new Date(Date.now() - 10_000) })),
+  );
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function hasExactly(arr: any[], count: number, predicate: (item: any) => boolean) {
+  let matchCount = 0;
+  for (const item of arr) {
+    if (predicate(item)) {
+      matchCount++;
+      if (matchCount > count) return false;
+    }
+  }
+  return matchCount === count;
+}
```

Comment:
```suggestion
// eslint-disable-next-line @typescript-eslint/no-explicit-any
function hasExactly(arr: any[], count: number, predicate: (item: any) => boolean) {
  return arr.filter(predicate).length === count;
}
```
minor: This version is a bit more compact and do the same thing although yours, especially with large arrays, is faster since it stops as soon as it exceeds the count.

Reply:
Oooo so good, thanks!